rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/EditDynamicParametersWindow.java

```diff
@@ -0,0 +1,161 @@
+
+        TextAreaItem previousValueItem = createTextItemPrefilledWithFileContent(plugin, pluginField);
+        formItemsReplacingNonTextualFormItems.add(previousValueItem);
+
+        return formItemsReplacingNonTextualFormItems;
+    }
+
+    private TextAreaItem createTextItemPrefilledWithFileContent(PluginDescriptor plugin,
+            PluginDescriptor.Field pluginField) {
+
+        TextAreaItem itemWithFileContent = new TextAreaItem(plugin.getPluginName() + pluginField.getName() +
+                                                            EDIT_FORM_ITEM_SUFFIX, pluginField.getName());
+
+        itemWithFileContent.setDefaultValue(pluginField.getValue());
+        return itemWithFileContent;
+    }
+
+    @Override
+    protected void handleAdditionalPolicyFormItems(Map<String, String> selectItemValues,
+            PluginDescriptor focusedPolicyPlugin) {
+        // do nothing
+    }
+
+    @Override
+    protected void handleAdditionalInfrastructureFormItems(Map<String, String> selectItemValues,
+            PluginDescriptor focusedInfrastructurePlugin) {
+        // do nothing
+    }
+
+    @Override
+    protected List<FormItem> modifyFormItemsAfterCreation(PluginDescriptor focusedInfrastructurePlugin,
+            PluginDescriptor focusedPolicyPlugin) {
+
+        List<PluginDescriptor.Field> dynamicFields = focusedPolicyPlugin.getConfigurableFields()
+                                                                        .stream()
+                                                                        .filter(PluginDescriptor.Field::isDynamic)
+                                                                        .collect(Collectors.toList());
+
+        List<PluginDescriptor.Field> dynamicIFields = focusedInfrastructurePlugin.getConfigurableFields()
+                                                                                 .stream()
+                                                                                 .filter(PluginDescriptor.Field::isDynamic)
+                                                                                 .collect(Collectors.toList());
+        dynamicFields.addAll(dynamicIFields);
+
+        List<FormItem> allFormItemsWithHiddenFields = new LinkedList<>();
+
+        for (FormItem formItem : this.allFormItems) {
+            allFormItemsWithHiddenFields.add(formItem);
+            if (dynamicFields.stream()
+                             .noneMatch(field -> formItem.getName()
+                                                         .equals(focusedPolicyPlugin.getPluginName() +
+                                                                 field.getName()) ||
+                                                 formItem.getName().equals(focusedInfrastructurePlugin.getPluginName() +
+                                                                           field.getName() + EDIT_FORM_ITEM_SUFFIX))) {
+
+                if (this.allFormItemsPerPlugin.keySet()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)